apps/data/src/tasks/data_upload_task.py

```
@@ -0,0 +1,450 @@
+    uploaded_at = datetime.now(timezone.utc)
+    all_rows: list[dict] = []
+    file_count = 0
+    error_count = 0
+
+    for path in matched_files:
+        try:
+            local_path = path.replace("dbfs:", "/dbfs") if path.startswith("dbfs:") else path
+            df = parser(local_path)
+            df = df.where(pd.notnull(df), None)
+            rows = df.to_dict(orient="records")
+            logger.info(f"Parsed {os.path.basename(path)}: {len(rows)} rows")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)